amida-tech/amida-auth-microservice
# Feature/orange 835 create user scopes

***I accidentally branched this off of https://github.com/amida-tech/amida-auth-microservice/pull/55 so we should review/merge that PR before this one***

#### What's this PR do?
This allows the scopes allowed to create users to be configured via environment variable. See README edits for details.

This *will* require `.env` changes in deployments!

#### Related JIRA tickets:
Orange-835

#### How should this be manually tested?
run `yarn install`
set `AUTH_SERVICE_REGISTRAR_SCOPES` to something like `["registrar"]`.
As the admin user, create a user that has the scope specified. This should succeed.
As that user create another lay-user. This should succeed.
As the lay-user try to create a user. This should fail.
As noone (no jwt in auth header) try to create a user. This should fail.

#### Any background context you want to provide?
This is useful where users who are not necessarily in need of full admin privileges, but do need the ability to provision other user accounts. (Like in 🍊)
#### Screenshots (if appropriate):
NA
